students/ptfbanks/Lesson09/obj_mailroom.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+# Lesson09 Object Mailroom exercise - ptfbanks
+
+
+class Donor:
+
+    def __init__(self, first_name, last_name, donations = None):
+        self.first = first_name
+        self.last = last_name
+        self.donations = donations
+
+    @property
+    def full_name(self):
+        return "{} {}".format(self.first, self.last)
+
+    def add_donation(self, amount):
+        return self.donations.append(amount)
+        
+    def total_donation(self):
+        return sum(self.donations)
+
+def name_input():
+    return input("Type the donor's full name: ")
+        
+def donation_input():
+    return float(input("Please enter the donation amount: "))
+
+letter = ("Dear {},\n"
+   "\t Thank you for your very kind donation(s) totaling $ {}.\n"
+   "It will be put to very good use.\n\n"
+   "\t\t Sincereley,\n"
+   "\t\t   -The Team")
+
+
+class DonorHistory:
+    def __init__(self, donors = None):
+        if donors is None:
+            self.donors = []
+        else:
+            self.donors = donors
+
+    def add_donor(self, donor):
+        self.donors.append(donor)
+
+    def get_all_donor_names(self):
+        return [donor.full_name for donor in self.donors]
+    
+    def thank_you(self):
+        donor_name = None
+        while not donor_name:
+            donor_name = name_input()
```

Comment:
you shouldn't have any `input` calls in your data classes (even though it's not used directly)